### PR TITLE
Remove storage connector

### PR DESCRIPTION
**Pull request by kteb**
This is a way to see the impact of the following proposal: #330 

As only mango & rdbms were using it. For me it make sense to remove it, and integrate it to the creation of the connection.

### CODE REVIEW

File: pkg/storage/mongo/mongo.go [hunk 1]
```diff
@@ -16,18 +16,19 @@ type ModuleStore struct {
...
-// that satisfies the Backend interface.  You must call
-// Connect() on the returned store before using it.
-func NewStorage(url string) *ModuleStore {
-	return &ModuleStore{url: url}
+// NewStorage returns a connected Mongo backed storage
+// that satisfies the Backend interface.
+func NewStorage(url string) (*ModuleStore, error) {
+	ms := &ModuleStore{url: url}
+	err := ms.connect()
+	return ms, err
```
Comment (arschles) [inline pkg/storage/mongo/mongo.go]: @kteb could you return `nil, err` if the call to `connect` failed?

Comment (robjloranger) [reply pkg/storage/mongo/mongo.go]: :+1:

File: pkg/storage/rdbms/test_suite.go [hunk 2]
```diff
@@ -14,12 +14,12 @@ type TestSuite struct {
 // NewTestSuite creates a common test suite
 func NewTestSuite(model *suite.Model) (storage.TestSuite, error) {
 	conn := model.DB
-	rdbmsStore := NewRDBMSStorageWithConn(conn)
+	rdbmsStore, err := NewRDBMSStorageWithConn(conn)
```
Comment (robjloranger) [inline pkg/storage/rdbms/test_suite.go]: can you check the error here and return it with a nil TestSuite instead? Then nil on last return.

File: pkg/storage/mongo/test_suite.go [hunk 3]
```diff
@@ -33,12 +33,12 @@ func newTestStore() (*ModuleStore, error) {
 		return nil, err
 	}
 
-	mongoStore := NewStorage(muri)
+	mongoStore, err := NewStorage(muri)
 	if mongoStore == nil {
```
Comment (marpio) [inline pkg/storage/mongo/test_suite.go]: Maybe just check the error instead?

**Comment by robjloranger** [comment]
Just curious, how did you merge the changes from master? It's making it look like you are adding them in this PR.

File: cmd/proxy/.env [hunk 4]
```diff
@@ -1 +1,3 @@
 ATHENS_MONGO_STORAGE_URL=mongodb://127.0.0.1:27017
+GO_ENV=test_postgres
```
Comment (michalpristas) [inline cmd/proxy/.env]: I know this is not the main focus of this PR. but can you please include
`TMPDIR=/tmp/` which is needed if the contributor does not have buffalo installed and 
`PATH=${PATH}:./bin` which is needed if somebody installs buffalo using our script

File: pkg/storage/mongo/mongo.go [hunk 5]
```diff
@@ -16,18 +16,24 @@ type ModuleStore struct {
...
-func NewStorage(url string) *ModuleStore {
-	return &ModuleStore{url: url}
+// NewStorage returns a connected Mongo backed storage
+// that satisfies the Backend interface.
+func NewStorage(url string) (*ModuleStore, error) {
+	ms := &ModuleStore{url: url}
+
+	err := ms.connect()
+	if err != nil {
+		return nil, err
```
Comment (michalpristas) [inline pkg/storage/mongo/mongo.go]: can you please
```
const op errors.Op = "fs.NewStorage"
return nil, errors.Op(op, err)
```

File: pkg/storage/rdbms/rdbms.go [hunk 6]
```diff
@@ -11,27 +11,26 @@ type ModuleStore struct {
...
-// connectionName
-func NewRDBMSStorageWithConn(connection *pop.Connection) *ModuleStore {
-	return &ModuleStore{
+// that satisfies the Storage interface.
+func NewRDBMSStorageWithConn(connection *pop.Connection) (*ModuleStore, error) {
+	ms := &ModuleStore{
 		conn: connection,
 	}
+	err := ms.connect()
+	return ms, err
```
Comment (michalpristas) [inline pkg/storage/rdbms/rdbms.go]: can you please
```
const op errors.Op = "fs.NewRDBMSStorageWithConn"
return nil, errors.Op(op, err)
```

File: scripts/test_unit.sh [hunk 7]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash
 
 # test_unit.sh
+
+source cmd/proxy/.env
```
Comment (michalpristas) [inline scripts/test_unit.sh]: if you include changes requested up there (.env) please make sure that `.env` is sourced before buffalo download starts. ideally at the beginning of `set-dev-env`
also there is a PR in focusing on this so you might get rid of it entirely as well #449

Comment (robjloranger) [reply scripts/test_unit.sh]: I think it's an unintentional result of a rebase, not a direct change from this PR

**Comment by kteb** [comment]
As it's kind of a big mess (because of my rebase, I don't want to take credit of other people work) I thought maybe to close this PR and recreate a new one without this issue  what do you think ?

File: pkg/storage/rdbms/rdbms.go [hunk 8]
```diff
@@ -10,28 +11,31 @@ type ModuleStore struct {
...
+	}
+	return ms, nil
 }
 
 // NewRDBMSStorageWithConn  returns a connected RDBMS Module Storage
-// that satisfies the Storage interface. You must call
-// Connect() on the returned store before using it.
-// connectionName
+// that satisfies the Storage interface.
 func NewRDBMSStorageWithConn(connection *pop.Connection) *ModuleStore {
```
Comment (kteb) [inline pkg/storage/rdbms/rdbms.go]: Not sure if this method should really stay here, it's only used in a test (in test_suite.go) not it sure if it's really needed

File: cmd/proxy/.env (hunk 4 above)
Comment (arschles) [reply cmd/proxy/.env]: @michalpristas would it be ok for us to do this in a follow up?

File: pkg/storage/mongo/test_suite.go (hunk 3 above)
Comment (arschles) [reply pkg/storage/mongo/test_suite.go]: +1, I feel like that would read nicer. @kteb can you do an `if err != nil` here instead?

File: pkg/storage/rdbms/rdbms.go (hunk 8 above)
Comment (arschles) [reply pkg/storage/rdbms/rdbms.go]: @kteb could the test use some other kind of module store? if so, it's up to you whether you'd like to remove it in this PR or wait for a follow-up. I usually do things like that in a follow up for what it's worth

File: cmd/proxy/.env (hunk 4 above)
Comment (michalpristas) [reply cmd/proxy/.env]: this was addressed in other PRs. please ignore my previous comment

File: pkg/storage/mongo/test_suite.go [hunk 9]
```diff
@@ -33,12 +33,12 @@ func newTestStore() (*ModuleStore, error) {
 		return nil, err
 	}
 
-	mongoStore := NewStorage(muri)
-	if mongoStore == nil {
+	mongoStore, err := NewStorage(muri)
+	if err != nil {
 		return nil, fmt.Errorf("Mongo storage is nil")
```
Comment (robjloranger) [inline pkg/storage/mongo/test_suite.go]: Just need to adjust the error message

File: pkg/storage/rdbms/rdbms.go (hunk 8 above)
Comment (arschles) [reply pkg/storage/rdbms/rdbms.go]: Follow-up: https://github.com/gomods/athens/issues/491

**Pull request merged**